File: FinanceApp.Shared/Dto/Loan/UpdateLoan.cs
using FinanceApp.Shared.Dto.Base;

namespace FinanceApp.Shared.Dto.Loan
{
    public class UpdateLoan : UpdateDto
    {
        public DateTime InitialDate { get; set; }
        public int MonthsPayment { get; set; }
        public string Name { get; set; }
        public double LoanValue { get; set; }
        public double InterestRate { get; set; }
        public int Type { get; set; }
    }
}

File: FlutterApp/finance_app/lib/models/loan/update_loan.dart
class UpdateLoan {
  String id;
  String name;
  DateTime initialDate;
  int monthsPayment;
  double loanValue;
  double interestRate;
  int type;

  UpdateLoan({
    required this.id,
    required this.name,
    required this.initialDate,
    required this.monthsPayment,
    required this.loanValue,
    required this.interestRate,
    required this.type,
  });

  Map<String, dynamic> toJson() => {
        'id': id,
        'name': name,
        'initialDate': initialDate.toString(),
        'monthsPayment': monthsPayment,
        'loanValue': loanValue,
        'interestRate': interestRate,
        'type': type,
      };
}
